projects/shared/service/search-bar-config.service.ts

```diff
@@ -0,0 +1,199 @@
+   * @param maxLengthSuggestion - number
+   * @return array of contribution types configuration
+   */
+  getConfig(isAdmin: boolean, viewcode?: string | undefined, maxLengthSuggestion?: number) {
+    this._isAdmin = isAdmin;
+    this._viewcode = viewcode;
+    this._maxLengthSuggestion = maxLengthSuggestion;
+
+    // If you add a new type, please specify it in the "shared-config.service.ts"
+    // file on contributionAgentTypes attribute
+    return [{
+      type: 'documents',
+      field: 'autocomplete_title',
+      maxNumberOfSuggestions: 5,
+      getSuggestions: (query: any, documents: any) => this.getDocumentsSuggestions(query, documents),
+      preFilters: viewcode !== undefined ? { view: viewcode } : {}
+    }, {
+      type: 'contributions',
+      field: 'autocomplete_name',
+      maxNumberOfSuggestions: 5,
+      getSuggestions: (query: any, persons: any) => this.getPersonsSuggestions(query, persons),
+      component: this,
+      preFilters: viewcode !== undefined ? { view: viewcode, type: 'bf:Person' } : { type: 'bf:Person' }
+    }, {
+      type: 'contributions',
+      field: 'autocomplete_name',
+      maxNumberOfSuggestions: 5,
+      getSuggestions: (query: any, organisations: any) => this.getOrganisationsSuggestions(query, organisations),
+      component: this,
+      preFilters: viewcode !== undefined ? { view: viewcode, type: 'bf:Organisation' } : { type: 'bf:Organisation' }
+    }
+    ];
+  }
+
+  /**
+   * Get persons suggestions
+   * @param query - string, string searched
+   * @param persons - search result records
+   * @return array - array of records formatted
+   */
+  getPersonsSuggestions(query: string, persons: any[]) {
+    return this._getContributionsSuggestions(
+      query, persons, 'persons', 'person direct links', 'user'
+    );
+  }
+
+  /**
+   * Get organisations suggestions
+   * @param query - string, string searched
+   * @param organisations - search result records
+   * @return array - array of records formatted
+   */
+  getOrganisationsSuggestions(query: string, organisations: any[]) {
+    return this._getContributionsSuggestions(
+      query, organisations, 'corporate-bodies', 'corporate bodies direct links', 'building'
+    );
+  }
+
+  /**
+   * Get documents suggestions
+   * @param query - string, string searched
+   * @param documents - search result records
+   * @return array - array of records formatted
+   */
+  getDocumentsSuggestions(query, documents) {
+    const values = [];
+    documents.hits.hits.map((hit: any) => {
+      let text = this._mainTitlePipe.transform(hit.metadata.title);
+      let truncate = false;
+      if (text.length > this._maxLengthSuggestion) {
+        truncate = true;
+        text = this._mainTitlePipe.transform(hit.metadata.title).substr(0, this._maxLengthSuggestion);
+      }
+      text = text.replace(new RegExp(this._escapeRegExp(query), 'gi'), `<b>${query}</b>`);
+      if (truncate) {
+        text = text + ' ...';
```

Comment:
'. . .' has an unicode string insteaf of 3 points :: '…'

Reply:
done